### PR TITLE
Rework the infrastructure to send messages via Slack

### PR DESCRIPTION
**Pull request by xrchz**

### CODE REVIEW

File: serverLib.sml [hunk 1]
```diff
@@ -369,21 +369,23 @@ structure GitHub = struct
     end
 end
 
+
+ok%
```
Comment (xrchz) [inline serverLib.sml]: delete this?

File: serverLib.sml [hunk 2]
```diff
@@ -369,21 +369,23 @@ structure GitHub = struct
...
-  val channel = "CCKESLBJQ"
-  val postMessage_endpoint = "https://slack.com/api/chat.postMessage"
+  val url = until_space (file_to_string "custom-uri")
+  val postMessage_endpoint = "https://hooks.slack.com/services/"
   fun postMessage_curl_cmd text = (curl_path,["--silent","--show-error",
-    "--header",String.concat["Authorization: Bearer ",token],
-    "--header","Content-type: application/json;charset=utf-8",
     "--request","POST",
+    "--header","Content-type: application/json;charset=utf-8",
     "--data",String.concat["{\"channel\":\"",channel,"\",",
```
Comment (xrchz) [inline serverLib.sml]: channel got deleted?

File: serverLib.sml (hunk 1 above)
Comment (HeikoBecker) [reply serverLib.sml]: Yes, Sorry that ended up in the file

File: serverLib.sml (hunk 2 above)
Comment (HeikoBecker) [reply serverLib.sml]: Yes. The bot can only send to the „worker-admins“ Channel as per the web configuration. Slack required picking a channel for web apps now.

Comment (xrchz) [reply serverLib.sml]: yeah but you're still using the variable that got deleted

File: serverLib.sml [hunk 3]
```diff
@@ -369,21 +369,20 @@ structure GitHub = struct
     end
 end
 
+
```
Comment (xrchz) [inline serverLib.sml]: let's remove this extra blank line

File: serverLib.sml [hunk 4]
```diff
@@ -369,21 +369,20 @@ structure GitHub = struct
...
-                           "\"text\":\"",text,"\"}"],
-    postMessage_endpoint])
+    "--header","Content-type: application/json;charset=utf-8",
+    "--data",String.concat["{\"text\":\"",text,"\"}"],
+    String.concat [postMessage_endpoint, url]])
   fun send_message text =
     let
       val text = String.translate (fn c => if c = #"\"" then "&quot;" else String.str c) text
       val cmd = postMessage_curl_cmd text
+      val _ = print cmd
```
Comment (xrchz) [inline serverLib.sml]: and remove this print too

**Pull request merged**